internal/compiler/diagnostics.rs

```diff
@@ -112,6 +114,25 @@ impl SourceFileInner {
         )
     }
 
+    pub fn text_range_to_file_line_column(
+        &self,
+        range: TextRange,
+    ) -> (String, usize, usize, usize, usize) {
+        let file_name = self.path().to_string_lossy().to_string();
+        let (start_line, start_column) = self.line_column(range.start().into());
+        let (end_line, end_column) = self.line_column(range.end().into());
+        (file_name, start_line, start_column, end_line, end_column)
+    }
+
+    pub fn text_size_to_file_line_column(
+        &self,
+        size: TextSize,
+    ) -> (String, usize, usize, usize, usize) {
```

Comment:
why return 4 usize. This is not a range.

Also wondering if we really need that when one can just call source_location.line_column an source_location.path() separately.